internal/logging/sentry_replacer.go

```diff
@@ -30,13 +32,18 @@ var replacement = []byte{'?'}
 
 type SentryReplacer struct {
 	buf    []byte
-	w      io.Writer
+	w      zerolog.LevelWriter
+	levels map[zerolog.Level]struct{}
 	re     *regexp.Regexp
 	mu     sync.Mutex
 	closed bool
 }
 
-func NewSentryReplacer(w io.Writer) *SentryReplacer {
+func NewSentryReplacer(w zerolog.LevelWriter, levels ...zerolog.Level) *SentryReplacer {
+	lvls := make(map[zerolog.Level]struct{}, len(levels))
+	for _, lvl := range levels {
+		lvls[lvl] = struct{}{}
+	}
```

Comment:
Where `lvls` is actually used?

Reply:
just to filter out the levels we do not want to filter on in the `WriteLevel`. I've copied this from https://github.com/archdx/zerolog-sentry/blob/a5e7c6e3cfc1dabe917a8ce22c4a7642a0c9bafd/writer.go#L78 but it might not be the best approach. I'd love to use the library internals instead, but it's not really possible.

Maybe we can just copy the library and merge the funcionality? :shrug: